website/database.py

```diff
@@ -227,30 +227,62 @@ def delete_class(self, Class):
         for student_id in Class.get ('students', []):
             Database.remove_student_from_class (self, Class ['id'], student_id)
 
-        CLASSES.delete({'id': Class ['id']})
-        PREFERENCES.delete({'id': Class['id']})
+        for customization in CUSTOMIZATIONS.get_many({'id': Class['id']}):
+            Database.remove_customizations_class(self, Class['id'], customization['level'])
+
+        CLASSES.delete({'id': Class['id']})
 
     def resolve_class_link(self, link_id):
         return CLASSES.get({'link': link_id})
 
-    def update_preferences_class(self, class_id, preferences):
-        preferences['id'] = class_id
-        PREFERENCES.create(preferences)
-
-    def get_preferences_class(self, class_id):
-        levels = []
-        preferences = PREFERENCES.get_many({'id': class_id})
-        for level in preferences:
-            levels.append(level)
-        temp = {}
-        for preference in preferences:
-            temp[preference['level']] = preference
-            temp[preference['level']].pop('level', None)
-        return temp
-
-    def get_level_preferences_class(self, class_id, level):
-        preferences = PREFERENCES.get_many({'id': class_id})
-        for preference in preferences:
-            if preference['level'] == level:
-                return preference
-        return None
\ No newline at end of file
+    def remove_customizations_class(self, class_id, level):
+        CUSTOMIZATIONS.delete({'id': class_id, 'level': level})
+
+    def update_customizations_class(self, level_customizations):
+        CUSTOMIZATIONS.put(level_customizations)
+        for customization in CUSTOMIZATIONS.get_many({'id': level_customizations['id']}):
+            if customization['level'] == (level_customizations['level']-1):
+                customization['hide_next_level'] = level_customizations['hide']
+                CUSTOMIZATIONS.put(customization)
+            elif customization['level'] == (level_customizations['level']+1):
+                customization['hide_prev_level'] = level_customizations['hide']
+                CUSTOMIZATIONS.put(customization)
+
+    def get_customizations_class(self, class_id):
+        customizations = {}
+        for customization in CUSTOMIZATIONS.get_many({'id': class_id}):
+            customizations[customization['level']] = customization
+        return customizations
+
+    def get_level_customizations_class(self, class_id, level):
+        try:
+            return CUSTOMIZATIONS.get({'id': class_id, 'level': level})
+        except:
+            return None
+
+    def get_student_restrictions(self, adventures, user, level):
```

Comment:
Does `adventures` here represent `all_adventures`, and are we filtering that list down?

Reply:
all adventures. Changed variable naming with last commit to make it more clear